### PR TITLE
Change json for endpoint actuator health

### PR DESCRIPTION
**Pull request by dbraquart**
1) All study services are now **optional: false**
2) The returned Json is changed to return all optional services, with its status. Ex:
```
[
  {
    "name": "security-analysis-server",
    "status": "UP"
  },
  {
    "name": "sensitivity-analysis-server",
    "status": "DOWN"
  },
  {
    "name": "shortcircuit-server",
    "status": "DOWN"
  },
  {
    "name": "dynamic-simulation-server",
    "status": "DOWN"
  },
  {
    "name": "voltage-init-server",
    "status": "UP"
  }
]
```

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -1407,12 +1407,12 @@ public ResponseEntity<Void> copyVoltageInitModifications(@PathVariable("studyUui
...
-    @Operation(summary = "Get all the available optional services")
-    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "List of available optional services")})
-    public ResponseEntity<List<String>> getUpOptionalServices() {
-        List<String> upServices = actuatorHealthService.getUpOptionalServices();
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(upServices);
+    @GetMapping(value = "/optional-services")
+    @Operation(summary = "Get all the optional services and their status")
+    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "List of optional services")})
+    public ResponseEntity<String> getOptionalServices() {
+        String optionalServicesStatus = actuatorHealthService.getOptionalServices();
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1414]: Why you do not use a DTO ?

File: src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java [hunk 2]
```diff
@@ -58,18 +59,42 @@ private SimpleClientHttpRequestFactory getClientHttpRequestFactory() {
...
+                .toList();
+
+        // merge all objects into a single one, to get:
+        //  <
+        //      "security-analysis-server": <
+        //          "status":"up"
+        //      >,
+        //      ...
+        //   >
+        ObjectNode mergedNode = objectMapper.createObjectNode();
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java:77]: Use a DTO

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/StudyController.java:1414]: well I dont see how to make the excepted Json using DTO...
Easy If I change the Json into:
```
[
  {
    "service": "security-analysis-server",
    "status": "up"
  },
  {
    "service": "short-circuit-server",
    "status": "down"
  },
 ...   
]
```

File: src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java [hunk 3]
```diff
@@ -22,18 +24,26 @@
...
 
 import static org.gridsuite.study.server.StudyConstants.DELIMITER;
 
 @Service
 public class ActuatorHealthService {
+
+    @Builder
+    @Getter
+    public static class ServiceStatusInfos {
+        private String service;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java:37]: Rename by name ?

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**